indexer/services/ender/src/scripts/dydx_tendermint_event_to_transaction_index.sql

```diff
@@ -0,0 +1,27 @@
+/**
+  Converts the event to its transaction index.
+  Parameters:
+    - event: The IndexerTendermintEvent object.
+  Returns: int.
+*/
+CREATE OR REPLACE FUNCTION dydx_tendermint_event_to_transaction_index(event jsonb) RETURNS int AS $$
+DECLARE
+    transaction_index_text text;
+    block_event_text text;
+BEGIN
+    transaction_index_text := jsonb_extract_path_text(event, 'transactionIndex');
+    block_event_text := jsonb_extract_path_text(event, 'blockEvent');
+
+    IF transaction_index_text IS NOT NULL THEN
+        RETURN transaction_index_text::int;
+    ELSIF block_event_text IS NOT NULL THEN
+        CASE block_event_text
+            WHEN '1' THEN RETURN -2;
+            WHEN '2' THEN RETURN -1;
```

Comment:
nit: can you include a comment to why '1' returns -2 and why '2' returns -1?

Reply:
done